### PR TITLE
[Story] Move upload buttons from "Create" button context menu to "Upload" button context menu

### PR DESCRIPTION
**Pull request by tibormichel1**
## Description

Upload file and upload folder options in modal menu are not shown if is clicked on upload button. If clicked on upload button, only upload folder and upload file options in context menu are shown

## Related Issue

 https://github.com/linagora/twake-drive/issues/226

## Motivation and Context
Upload button before was only able to upload files, now it makes more sense because user can select also folders to upload after clicking on upload button (no need to click create button = more intuitive approach).

## How Has This Been Tested?

Tested by myself (manual testing), checked ui as well as functionality.

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added the Signed-off-by statement at the end of my commit message.

### CODE REVIEW

File: tdrive/frontend/src/app/views/client/body/drive/modals/create/index.tsx
Comment (MontaGhanmy) [inline tdrive/frontend/src/app/views/client/body/drive/modals/create/index.tsx:1]: Please duplicate the 'create' modal into a dedicated 'upload' modal or, ideally, create a generic 'modal' component and extend both 'create' and 'upload' from it. The current structure of combining both functionalities within a single modal complicates code comprehension and maintenance.

**Comment by MontaGhanmy** [comment]
Hi @tibormichel1, thank you for your interest in Twake-drive. I just left you some feedback to fix before merging your PR. please let me know if anything is not clear.

File: tdrive/frontend/src/app/views/client/body/drive/browser.tsx
Comment (MontaGhanmy) [inline tdrive/frontend/src/app/views/client/body/drive/browser.tsx:1]: Please make sure the modal closes once the file/folder is uploaded.

**Review by MontaGhanmy — CHANGES_REQUESTED** [review]

**Review by MontaGhanmy — COMMENTED** [review]
@tibormichel The only remaining issue is with closing the modal once the upload is finished.

**Review by shepilov — APPROVED** [review]

**Review by ericlinagora — COMMENTED** [review]
The CreateModalOption duplication is a bit odd and has an odd name for a component but not really an issue

**Pull request merged**